src/views/dashboard/User.jsx

```diff
@@ -1,33 +1,26 @@
-import React, { useEffect } from "react";
+import React, { useState, useEffect } from "react";
 import { Query } from "react-apollo";
 import { GET_USER } from "../../graphql-requests/queries";
 import "../home/Home.css";
 import TextField from "@material-ui/core/TextField";
 //import { makeStyles } from "@material-ui/core/styles";
 
-// const useStyles = makeStyles(theme => ({
-//    root: {
-//       "& > *": {
-//          margin: theme.spacing(1),
-//          width: "35ch",
-//       },
-//    },
-// }));
-
-const user_id = localStorage.getItem("user_id");
-
 const User = props => {
+   const [user_id, setUser] = useState("");
    useEffect(() => {
-      console.log(GET_USER);
-   }, []);
+      let user_id = localStorage.getItem("user_id");
+      if (user_id) {
+         setUser(user_id);
+      }
+   }, [user_id]);
```

Comment:
I feel like this should cause an infinite loop. I may be wrong, can someone explain this one to me?
I ask because, this `useEffect` should run whenever `user_id` changes, but it changes `user_id` within the `useEffect`.

Reply:
I've removed it as a precaution and everything runs fine I believe the reason it does not cause an infinite loop is here @Hook API. "You can tell React to skip applying an effect if certain values haven’t changed between re-renders. To do so, pass an array as an optional second argument to useEffect" which only changes once and is held in state, but I could be wrong